data_structures/src/chain.rs

```diff
@@ -492,7 +564,6 @@ pub struct ValueTransferInput {
 pub struct CommitInput {
     pub transaction_id: Hash,
     pub output_index: u32,
-    pub reveal: Vec<u8>,
     pub nonce: u64,
```

Comment:
Does it make sense to also remove `nonce` from `CommitInput`?
The nonce and the reveal are both needed to validate the commitment hash, so they probably should be stored together.

Reply:
Speaking with @aesedepece, a future Inputs, Outputs refactor is probably coming

Reply:
IMO:
- we should have no input types at all
- `inputs` should always be plain `OutputPointer`s
- the `signatures` field in `Transaction` should be renamed to `proofs` as it can contain other info than signatures, e.g. `Reveal` transactions are spent by `Tally` without providing a signature, and so do `Commitment` with `DataRequest`
- generally speaking, the point of `proofs` is proving the ability to spend specific UTXOs
- conversely, every piece of data with the mentioned purpose should be displaced to `proofs`
- when checking whether an input + proof pair is valid, the validation function should:
  - operate only over the pointed output  `validate_spend(output: Output, proof: Proof)`
  - pivot over the type of `output` and apply different validation logic
  - therefore different `output` types require `proof` to provide different data—which makes me think that there are different types of `proof`, e.g. `Proof::P2PKH(KeyedSignature)`